libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -161,6 +161,40 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
         )
     }
 
+    _LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_INLINE_VISIBILITY
```

Comment:
I am not sure whether we would want to unconditionally expose this

Reply:
This is also lacking test coverage which should go into `libcudacxx/.upstream-tests/test/cuda/barrier`

Reply:
Thanks for the pointer to the tests. I am not sure what you mean by unconditionally expose this. Do you mean only expose for specific device architectures or only expose on device? 

If we want the exposure to be more limited, should I use other macros?

Reply:
I meant to ifdef it out if the architecture does not support it

Reply:
That is possible. From a user's point of view I am not sure that would be the best experience. Example: user uses arrive_tx and compiles with -arch sm_80:

- When ifdefed out: the user gets an error that arrive_tx is not found. 
- When NVtargeted inside the function: users gets a static assert that arrive_tx is not supported on sm_80.

I am sure there are other considerations I am not taking into account. Let me know if so, and I can change to ifdef the function out.

Reply:
I though you meant to ifdef out the entire function when the architecture does not support it. I went down this path and I just cannot get it to work with NV_IF_TARGET. Is that what you intended, and if so, how do I ifdef out an entire function in a reliable way?